nixos/modules/services/system/nscd.nix

```diff
@@ -59,10 +59,9 @@ in
           '';
 
         restartTriggers = [
-          config.environment.etc.hosts.source
           config.environment.etc."nsswitch.conf".source
           config.environment.etc."nscd.conf".source
-        ];
+        ] ++ optional config.networking.enable config.environment.etc.hosts.source;
```

Comment:
why is this change required?

Reply:
Fair question, I should probably have called that out. It's because if networking is disabled, `/etc/hosts` doesn't get created at all, and `config.environment.etc.hosts` is undefined. But nscd is still usable without network so it shouldn't get automatically disabled entirely.

Thanks for reviewing!